lib/client/api.go

```diff
@@ -668,11 +668,26 @@ func WithBeforeLoginHook(fn func() error) RetryWithReloginOption {
 	}
 }
 
+// IsErrorResolvableWithRelogin returns true if relogin is attempted on `err`.
 func IsErrorResolvableWithRelogin(err error) bool {
-	// Assume that failed handshake is a result of expired credentials.
-	return utils.IsHandshakeFailedError(err) || utils.IsCertExpiredError(err) ||
-		trace.IsBadParameter(err) || trace.IsTrustError(err) ||
-		keys.IsPrivateKeyPolicyError(err) || IsNoCredentialsError(err)
+	// Ignore any failures resulting from RPCs.
+	// These were all materialized as status.Error here before
+	// https://github.com/gravitational/teleport/pull/30578.
+	var remoteErr *interceptors.RemoteError
+	if errors.As(err, &remoteErr) {
+		return false
```

Comment:
@codingllama should we treat all remote errors equally here?
We can get `Original Error: *interceptors.RemoteError access denied: client credentials have expired, please relogin.` for which we return `false` but it seems that for this particular case it should be `true`? 

I ask because I'm working on adding a client cache to Connect, and this means that the client no longer checks the user cert before making call, but instead it has to rely on errors from the server.

The original comment https://github.com/gravitational/teleport/pull/38202#discussion_r1497181659.

Reply:
>That means finding the client-side callsite for GenerateUserCerts, inspecting the response and marking it as retriable accordingly.

So we would have to wrap methods in `api/client/client.go` manually? Or could we use the interceptor (where we add `RemoteError`) and check for 'client credentials have expired' there?

Reply:
Wrapping client.go manually is the best choice, imo, as there is no loss of context and little chance of false-positives. That's how I think this should have started.

For a "generic" place I would do it here, not in the interceptor.